Add function that checks if function id represents a clinit.

jeannielynnmoulton opened the pull request:
A utility for checking whether the supplied function id corresponds to a
clinit.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Add function that checks if function id represents a clinit.

A utility for checking whether the supplied function id corresponds to a
clinit.

thomasspriggs reviewed (approved):
Unit test, as well please. You know it makes sense :)

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7f1830eb4f1cc2f96fd1c210c58cedfe7d2c3b12).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113106427

LAJW commented on jbmc/src/java_bytecode/java_static_initializers.cpp:
```diff
@@ -73,6 +73,14 @@ bool is_clinit_wrapper_function(const irep_idt &function_id)
+/// Check if function_id is a clinit
+/// \param function_id: some function identifier
+/// \return true if the passed identifier is a clinit
+bool is_clinit_function(const irep_idt &function_id)
```
🚫 Function not used.

smowton reviewed (changes requested):
Indeed, no unused code in the cbmc codebase please :)

Commit:
Unit test for is_clinit_function

Shows that is_clinit_function returns true iff input has method signature with a suffix that matches that of clinit

jeannielynnmoulton commented:
@smowton No problem, added.

smowton commented:
Add to https://github.com/diffblue/cbmc/blob/develop/jbmc/unit/Makefile

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c227ab5ca65bb56d2099e5b0214fed9d9b1e3af5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113421780
